nntrainer/dataset/databuffer.h

```diff
@@ -78,11 +79,49 @@ class DataBuffer : public ml::train::Dataset {
    */
   virtual int init();
 
+  /**
+   * @brief prepare iteration a head of time with a dedicated worker. The
+   * iteration prepared can be retrieved with @a fetch();
+   * @remark the batch dimension of input_dims / label_dims must be same for
+   * all.
+   * @param input_dims dimension of input_dims
+   * @param label_dims dimension of label_dims
+   * @return std::future<std::shared_ptr<BatchQueue>> Buffer Queue object,
+   * release this pointer after calling @a fetch() is done to invalidate
+   * subsequent call of @a fetch()
+   */
+  std::future<std::shared_ptr<BatchQueue>>
+  startFetchWorker(const std::vector<TensorDim> &input_dims,
+                   const std::vector<TensorDim> &label_dims);
+
+  /**
+   * @brief Get the Iteration object
+   * @note  the first element of returned Iteration denotes whether current
+   * epoch has ended.
+   *
+   * @throw std::invalid_argument if @a startFetchWorker hasn't been called or
+   * the return value of startFetchWorker has been invalidated.
+   * @return std::unique_ptr<DataProducer::Iteration> iteration
+   */
+  std::unique_ptr<DataProducer::Iteration> fetch();
+
+  /**
+   * @brief Get the Generator object
+   * @remark the batch dimension of input_dims / label_dims must be same for
+   * all.
+   *
+   * @param input_dims dimension of input_dims
+   * @param label_dims dimension of label_dims
+   * @return DataProducer::Generator which generates an iteration
+   */
+  DataProducer::Generator batcher(const std::vector<TensorDim> &input_dims,
```

Comment:
can you please elaborate on this? the name doesn't match with the brief.
Also, startFetchWorker() wraps around this, so maybe this is intended for private?

Reply:
This is sync method for the databuffer so intended to be a public method.

Later, if we ever going to support `model->forwarding` or equivalent, this will be needed.

Please refer to below, meanwhile I'll elaborate more :)

https://github.com/nnstreamer/nntrainer/blob/1da132aa82e08f5a3c49177ca42d6e4eb656747c/test/unittest/datasets/unittest_databuffer.cpp#L19-L34

Reply:
Then you might consider renaming to something like `getGenerator()` etc.

Reply:
I'll rename it to somewhat similar to the suggestion on the future work :)